src/Datadog.Trace.ClrProfiler.Managed/Instrumentation.cs

```diff
@@ -43,21 +50,53 @@ public static bool ProfilerAttached
         /// </summary>
         public static void Initialize()
         {
-            try
+            if (Interlocked.Exchange(ref _firstInitialization, 0) == 1)
```

Comment:
NIT: You could invert the condition to reduce nesting

Reply:
fixed in 66935a09